willow/lib/sufia/notifications/subscribers/build_message.rb

```diff
@@ -5,62 +5,278 @@ module Notifications
-                    objectOrganisationRole: [],
-                    objectPreservationEvent: [],
-                    objectFile: [],
+                    objectUuid: objectUuid,
+                    objectTitle: objectTitle,
+                    objectPersonRole: objectPersonRole,
+                    objectDescription: objectDescription,
+                    objectRights: objectRights,
+                    objectDate: objectDate,
+                    objectKeywords: objectKeywords,
+                    objectCategory: objectCategory,
+                    objectResourceType: objectResourceType,
+                    objectValue: objectValue,
+                    objectIdentifier: objectIdentifier,
+                    objectRelatedIdentifier: objectRelatedIdentifier,
+                    objectOrganisationRole: objectOrganisationRole,
+                    objectPreservationEvent: objectPreservationEvent,
+                    objectFile: objectFile,
                   }
               }
           }
+
+          # puts JSON.pretty_generate(msg)
+          msg
         end
 
         private
+
+        ##########################
+        # message header
+        ##########################
+
+        def messageId
+          SecureRandom.uuid
+        end
+
+        def messageClass
+          'Event'
+        end
+
         def messageType
-          case @event
-            when 'create_work.sufia'
-              'CREATE'
-            when 'update_work.sufia'
-              'UPDATE'
-            when 'delete_work.sufia'
-              'DELETE'
+          "#{@event}.#{@curation_concern_type}"
+        end
+
+        def messageTimings
+          { publishedTimestamp: DateTime.now.rfc3339 }
+        end
+
+        def messageSequence
+          {
+              sequence:SecureRandom.uuid,
+              position: 1,
+              total: 1
+          }
+        end
+
+        def version
+          '0.0.1-SNAPSHOT'
+        end
+
+        ##########################
+        # message body
+        ##########################
+
+        def objectUuid
+          @object.id
+        end
+
+        def objectTitle
+          @object.title.join('; ')
+        end
+
+        def objectPersonRole
+          case
+            when @curation_concern_type.eql?(Work)
+              @object.creator.map{|c|
+                {
+                    person: {
+                        personGivenName: c
+                    }
+                }
+              }
+            when [Dataset, Article].include?(@curation_concern_type) && @event != 'destroy_work.sufia'
+              @object.creator_nested.map{|c|
+                {
+                  person: {
+                         personGivenName: (c.first_name + c.last_name).join(' '),
+                         personIdentifier: c.orcid.map {|o|
+                           {
+                               personIdentifierValue: o,
+                               personIdentifierType: UNKNOWN_ID
```

Comment:
hi @anusharanganathan, the schema specifies that the `personIdentifierType` is a number, not a string. Do you know where it might be defined? See https://github.com/JiscRDSS/rdss-message-api-docs/blob/master/messages/metadata/create/request.json